### ISSUE TITLE
WC-22 - Implement MicroProfile OpenAPI 1.1.1 in Open Liberty

### ISSUE THREAD

**Issue by arturdzm** [OpenLiberty/open-liberty#6154]
Implementation of the MicroProfile OpenAPI 1.1.1.   See release notes for more information (https://github.com/eclipse/microprofile-open-api/releases/tag/mp-openapi-1.1.1), but this was mainly a maintenance release.  No real changes to the programming model.  The only worthy change is the addition of `PATCH`, which comes from JAX-RS 2.1

**Comment by alexmulholland** [comment]
Will this include the fix/implementation of file upload support in the OpenAPI UI ?
https://github.com/swagger-api/swagger-ui/issues/3784

**Comment by arthurdm** [comment]
@arturdzm - can we please ensure we include the UI fix mentioned above?

**Comment by arturdzm** [comment]
@alexmulholland @arthurdm The schedule for 19.0.0.3 is tight, but I will try to fit it in. 
I was thinking doing full UI refresh for 19.0.0.4, as the UI is not tied to the spec

**Comment by arthurdm** [comment]
Please note that 19.0.0.4 is _targeted_ for April, so not too far. =)

**Comment by marikaj123** [comment]
@arturdzm - Focal point owners started assessing the features going into 19003.  In case it was decided to make 19003 then please add the tags target:ga and FAT Complete. FOcal point owners need at least 2 weeks before March 22 to assess and complete their work.

**Comment by arturdzm** [comment]
- [ ] Implement changes to interfaces, add new feature. https://github.com/OpenLiberty/open-liberty/pull/6819 
- [ ] Add TCK 1.1 FAT #6851 
- [ ] GA Feature, #6851

**Comment by arturdzm** [comment]
New MP OpenAPI TCK 1.1 tests new changes to the API interfaces.
Existing FAT shared between mpOpenAPI-1.0 and mpOpenAPI-1.1 tests common code

**Comment by dave-waddling** [comment]
Hi @arturdzm , I've created and attached the Feature Test Summary (#6852) issue required for FAT Focal Approval. 

Given that #6851 is not merged can you confirm whether this issue is actually "FAT complete"?

**Comment by marikaj123** [comment]
@arturdzm - Does the feature have English files that require translation?

**Comment by arturdzm** [comment]
@marikaj123 

No, we are using same messages from before shared between 2 features.

For feature description, there is this this file which is a copy from mpOpenAPI-1.0.

https://github.com/OpenLiberty/open-liberty/blob/integration/dev/com.ibm.websphere.appserver.features/visibility/public/mpOpenAPI-1.1/resources/l10n/com.ibm.websphere.appserver.mpOpenAPI-1.1.properties

Should I copy the rest of translated files from mpOpenAPI-1.0 feature

**Comment by skasund** [comment]
L2 has requested STE slides for this feature.  The STE template can be found at the link below: 

https://ibm.box.com/s/1an42g7zdgmaj84w7dft0indqfgi8ffm

Please upload the completed slides to the same STE Archive BOX folder.  Thanks!

**Comment by marikaj123** [comment]
@arturdzm - Please provide the link to the AHA feature.

**Comment by atosak** [comment]
https://bigblue.aha.io/features/WC-22

**Comment by gscottj** [comment]
Accessibility Approval Comment:  
Based on the description provided, it appears that this feature does not change any end-user user interfaces that would require accessibility compliance testing. End user user-interfaces include GUI, CLI and documentation that users are expected to interact with to accomplish tasks or goals. If this is not correct, then let me know that required accessibility verification testing of the UI will be performed before it is shipped and what are the results from those tests.

**Comment by tam512** [comment]
approved SVT since no SVT needed

**Comment by skasund** [comment]
No STE is required.  I've approved the feature.

**Comment by donbourne** [comment]
**Serviceability Approval Comment** - Please answer the following questions for serviceability approval:

1.  WAD -- does the WAD identify the most likely problems customers will see and identify how the feature will enable them to diagnose and solve those problems without resorting to raising a PMR?  Have these issues been addressed in the implementation?

2.  Test and Demo -- As part of the serviceability process we're asking feature teams to test and analyze common problem paths for serviceability and demo those problem paths to someone not involved in the development of the feature (eg. L2, test team, or another development team).  
a) What problem paths were tested and demonstrated? 
b) Who did you demo to?
c) Do the people you demo'd to agree that the serviceability of the demonstrated problem scenarios is sufficient to avoid PMRs for any problems customers are likely to encounter, or that L2 should be able to quickly address those problems without need to engage L3?

3.  SVT -- SVT team is often the first team to try new features and often encounters problems setting up and using them.  Note that we're not expecting SVT to do full serviceability testing -- just to sign-off on the serviceability of the problem paths they encountered.
a) Who conducted SVT tests for this feature?
b) Do they agree that the serviceability of the problems they encountered is sufficient to avoid PMRs, or that L2 should be able to quickly address those problems without need to engage L3?  

4.  Which L2 / L3 queues will handle PMRs for this feature?  Ensure they are present in the contact reference file and in the queue contact summary, and that the respective L2/L3 teams know they are supporting it.  Ask Don Bourne if you need links or more info.

**Comment by arturdzm** [comment]
@donbourne 
For 1 and 2:
See https://github.com/OpenLiberty/open-liberty/issues/2292

The serviceability messages and tracing is shared between mpOpenAPI-1.0 and mpOpenAPI-1.1 features and located in com.ibm.ws.microprofile.openapi project
There was no need to add new messages, as the main change was addition of OpenAPI models for 1.1 specification. 

3. SVT was not required for this feature

4. WAS L3: Liberty JMX REST Connector

**Comment by sabolo** [comment]
Waiting for results of the next cross platform testing run to be able to verify the new tests.

**Comment by dave-waddling** [comment]
Feature Test Summary looks good and testing is looking good, adding the FAT Focal Approval.

**Comment by arturdzm** [comment]
Delivered in 19.0.0.3
